server/aws-lsp-codewhisperer/src/shared/localProjectContextController.ts

```diff
@@ -140,11 +141,25 @@ export class LocalProjectContextController {
                     `index worker thread count: ${indexWorkerThreads}`
             )
 
+            // build index if vecLib was initialized but indexing was not enabled before
+            if (this._vecLib) {
+                if (enableIndexing && !this._isIndexingEnabled) {
+                    void this.buildIndex()
+                }
+                this._isIndexingEnabled = enableIndexing
+                return
+            }
+
+            // initialize vecLib and index if needed
+            this._isIndexingEnabled = enableIndexing
```

Comment:
Shouldn't this assignment happen after we attempt to build the index? Otherwise a failure in the import or start calls would effectively prevent us from subsequently building the index.

Reply:
updated